From 7ebf02af2e90f03e0dbd0e18b8b3164f372fb97c Mon Sep 17 00:00:00 2001
From: Johannes Schultz <sagamusix@openmpt.org>
Date: Sat, 7 Apr 2018 16:04:51 +0000
Subject: [PATCH] [Fix] Possible out-of-bounds read when computing length of
 some IT files with pattern loops (OpenMPT: formats that are converted to IT,
 libopenmpt: IT/ITP/MO3), caught with afl-fuzz.

git-svn-id: https://source.openmpt.org/svn/openmpt/trunk/OpenMPT@10027 56274372-70c3-4bfc-bfc3-4c3a0b034d27
---
 soundlib/Snd_fx.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/soundlib/Snd_fx.cpp b/soundlib/Snd_fx.cpp
index c623da591f..29f93c1de1 100644
--- a/soundlib/Snd_fx.cpp
+++ b/soundlib/Snd_fx.cpp
@@ -1204,7 +1204,8 @@ std::vector<GetLengthType> CSoundFile::GetLength(enmGetLengthResetMode adjustMod
 			if(GetType() == MOD_TYPE_IT)
 			{
 				// IT pattern loop start row update - at the end of a pattern loop, set pattern loop start to next row (for upcoming pattern loops with missing SB0)
-				for(CHANNELINDEX nChn = 0; nChn < GetNumChannels(); nChn++)
+				pChn = playState.Chn;
+				for(CHANNELINDEX nChn = 0; nChn < GetNumChannels(); nChn++, pChn++)
 				{
 					if((pChn->rowCommand.command == CMD_S3MCMDEX && pChn->rowCommand.param >= 0xB1 && pChn->rowCommand.param <= 0xBF))
 					{
